Python3 part4

joshua-cogliati-inl opened the pull request:
--------
Pull Request Description
--------
##### What issue does this change request address? 
#71 

##### What are the significant changes in functionality due to this change request?
Better python 3 support.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/249171/) on 13c2ff8 : canceled by @joshua-cogliati-inl

will fail.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/249843/) on 73f870f : canceled by @joshua-cogliati-inl

will fail.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/249975/) on bb36a46 : canceled by @joshua-cogliati-inl

will fail.

moosebuild commented:
Job [Test mac](https://civet.inl.gov/job/249973/) on bb36a46 : canceled by @joshua-cogliati-inl

will fail.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/250003/) on 9473518 : canceled by @joshua-cogliati-inl

moosebuild commented:
Job [Test mac](https://civet.inl.gov/job/250001/) on 9473518 : canceled by @joshua-cogliati-inl

joshua-cogliati-inl commented:
#813 and #820 have been merged, so as soon as the tests pass this is ready to merge.

joshua-cogliati-inl commented:
This includes adding Parallel Python original files, so should probably not be squashed.

moosebuild commented:
Job [Test Fedora 27](https://civet.inl.gov/job/250089/) on 5558311 : invalidated by @joshua-cogliati-inl

alfoa commented on framework/PostProcessors/TopologicalDecomposition.py:
```diff
@@ -32,6 +33,18 @@
+  """
+  if sys.version_info.major > 2:
+    return s
+  return s.encode('ascii')
```
shouldn't we put this in the utils.py module so other modules can use it in case needed?

joshua-cogliati-inl replied:
Actually, I can remove it in all three cases by removing from __future__ import unicode_literals.

alfoa commented on framework/Samplers/AdaptiveDynamicEventTree.py:
```diff
@@ -375,6 +376,7 @@ def localStillReady(self,ready): #, lastOutput= None
     if self.startAdaptive:
       #if self._endJobRunnable != 1: self._endJobRunnable = 1
       data = self.lastOutput.asDataset()
+      print(self.lastOutput, data)
```
remove pls

joshua-cogliati-inl replied:
Removed.

alfoa commented on framework/Samplers/LimitSurfaceSearch.py:
```diff
@@ -43,6 +44,16 @@
+  """
+  if sys.version_info.major > 2:
+    return s
+  return s.encode('ascii','ignore')
```
as above. Place this in the utils.py so we are not redundant

joshua-cogliati-inl replied:
Removed.

alfoa commented on framework/SupervisedLearning/ARMA.py:
```diff
@@ -42,6 +42,18 @@
+  """
+  if sys.version_info.major > 2:
+    return s
+  return s.encode('ascii')
```
as above

alfoa replied:
+ the docstrings is actually copied/pasted

joshua-cogliati-inl replied:
Removed.

alfoa commented on framework/contrib/pp3/pp.py:
```diff
@@ -0,0 +1,886 @@
+    import user
+except ImportError:
+    user = types  # using as a stub
+import cloudpickle as pickle
```
Josh, can you tag (with a comment) the portions of the library we modified for making it work for us?

joshua-cogliati-inl replied:
I added ```#RAVEN CHANGE``` comments to the changed parts.

joshua-cogliati-inl commented:
If tests pass, I will clean up the commit history.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/251565/) on 4ed64b0 : canceled by @joshua-cogliati-inl

will fail.

joshua-cogliati-inl commented:
History cleaned up, please don't squash. The original history is now at: https://github.com/idaholab/raven/tree/python3_part4_orig  (The last three commits on it are the changes since the last review)

moosebuild commented:
Job [Test Ubuntu 16](https://civet.inl.gov/job/251679/) on 7cd909c : invalidated by @joshua-cogliati-inl

occ

moosebuild commented:
Job [Test OpenSUSE Leap 15](https://civet.inl.gov/job/251677/) on 7cd909c : invalidated by @joshua-cogliati-inl

occ

moosebuild commented:
Job [Test CentOS 7](https://civet.inl.gov/job/251963/) on afd0535 : invalidated by @joshua-cogliati-inl

wangcj05 commented:
@joshua-cogliati-inl Could you attach the test results for the python3? In this case, other PR can be tested with python3 and compared with your test results to make sure any new PR will not break currently passed tests. In addition, I think @alfoa had made a scripts in the past to compare the two test outputs to report the diff of these two test results, which can be used for checking our new PR.

joshua-cogliati-inl commented:
[test_out20.txt](https://github.com/idaholab/raven/files/2540272/test_out20.txt)
Attached Python 3 tests results.

moosebuild commented:
Job [Test CentOS 7](https://civet.inl.gov/job/251963/) on afd0535 : invalidated by @joshua-cogliati-inl

joshua-cogliati-inl commented:
Tests now pass.

alfoa commented:
PR can be merged.
